packages/flutter_tools/lib/src/build_system/file_hash_store.dart

```diff
@@ -88,20 +88,31 @@ class FileHashStore {
   /// Read file hashes from disk.
   void initialize() {
     printTrace('Initializing file store');
-    if (!_cacheFile.existsSync()) {
+    final File cacheFile = fileSystem.file(_cachePath);
+    if (!cacheFile.existsSync()) {
       return;
     }
-    final List<int> data = _cacheFile.readAsBytesSync();
+    List<int> data;
+    try {
+      data = cacheFile.readAsBytesSync();
+    } on FileSystemException catch (err) {
+      printError(
+        'Failed to read file store at ${cacheFile.path} due to $err.\n'
+        'Build artifacts will not be cached.'
```

Comment:
Consider suggesting to manually delete the file so that caching doesn't also fail the next time.

Reply:
Done